pifs/holdem_poker_pif.py

```diff
@@ -0,0 +1,222 @@
+
+`LatherBot in`
+
+I will check your karma and deal your cards if you qualify.
+
+This PIF will close in {} hour(s) or when I run out of unique hands (~1000).  
+At that time, I will determine the winner and notify the PIF's creator.
+
+LatherBot documentation can be found in [the wiki](https://www.reddit.com/r/Wetshaving/wiki/latherbot)
+
+The flop cards will be drawn now while the turn and river cards will be revealed at the end of the PIF.
+
+You will be dealt a unique combination of two hole cards. IE while two players may be dealt the Queen of Hearts, 
+only one player will ever be dealt the Queen of Hearts + the King of Hearts.
+
+Per Texas Hold'em rules the player who makes the best five card poker hand out of their hole cards, the flop, turn and river cards will win.
+In the event of a tie a winner will be selected randomly from the tied players. 
+
+Your three flop cards are {} {} and {}
+
+
+Good luck!
+"""
+
+entry_template = """
+{} drew {} and {} as their hole cards
+
+Good luck!
+"""
+
+winner_template = """
+The PIF is over!
+
+The turn card was {}
+The river card was {}
+
+u/{} has won with {} 
+({})
+
+{}
+"""
+
+
+class HoldemPoker(BasePIF):
+
+    def __init__(self, postId, authorName, minKarma, durationHours, endTime, pifOptions={}, pifEntries={},
+                 karmaFail={}):
+        logging.debug('Building holdem poker PIF [%s]', postId)
+
+        if len(pifOptions) < 1:
+            deck = poker_util.new_deck()
+            flop_cards = list()
+            for i in range(3):
+                card = poker_util.deal_card(deck)
+                flop_cards.append(card)
+
+            flop_cards = poker_util.order_cards(flop_cards)
+
+            river_card = poker_util.deal_card(deck)
+            turn_card = poker_util.deal_card(deck)
+            # remove 3 cards (ie never dealt) so these dont appear as hold cards
+            # and so users cant infer river and flop cards from looking at all dealt hands
+            for i in range(3):
+                poker_util.deal_card(deck)
+
+            pifOptions['Deck'] = deck
+            pifOptions['FlopCards'] = flop_cards
+            pifOptions['RiverCard'] = river_card
+            pifOptions['TurnCard'] = turn_card
+            pifOptions['ExtraInfo'] = ''
+
+            remaining_cards = list()
+            while True:
+                try:
+                    remaining_cards.append(poker_util.deal_card(deck))
+                except IndexError:
+                    break
+
+            # make a set of possible two card combinations for dealing
+            # note that two users can be dealt the same card, but not the exact same combination of cards
+            pifOptions['hands'] = itertools.combinations(remaining_cards, 2)
+            random.shuffle(pifOptions['hands'])
+
+        BasePIF.__init__(self, postId, authorName, 'poker', minKarma, durationHours, endTime, pifOptions, pifEntries,
```

Comment:
Need to fix the pif type here to be something other than 'poker'.
```suggestion
        BasePIF.__init__(self, postId, authorName, 'holdem-poker', minKarma, durationHours, endTime, pifOptions, pifEntries,
```